cmd/prometheus/main.go

```diff
@@ -186,6 +189,9 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			case "promql-per-step-stats":
 				c.enablePerStepStats = true
 				level.Info(logger).Log("msg", "Experimental per-step statistics reporting")
+			case "auto-gomaxprocs":
+				c.enableAutoGOMAXPROCS = true
+				level.Info(logger).Log("msg", "Automatically set GOMAXPROCS to cpu.limits when running in container.")
```

Comment:
If this feature flag is enabled, but Prometheus is not running in a container, will the `maxprocs.Set` call (below) fail and emit a warning?

Reply:
It'll not fail not emit warning, it'll leave the default which is equal to number of CPUs.

I'm not sure under which conditions would Set fail. I've added warning for user to know why the GOMAXPROCS is not set when this feature is enabled.